internal/trafficlogs/logger.go

```diff
@@ -51,24 +53,32 @@ func NewFileBased() *FileBasedLogger {
 	return tl
 }
 
-func (l *FileBasedLogger) Log(req *http.Request, reqBody string, resp *http.Response, respBody string) error {
-	l.lock.Lock()
-	defer l.lock.Unlock()
+// LogToFiles takes a record containing request and response information and tries to write it into the files
+// created by this logger.
+func (l *FileBasedLogger) LogToFiles(record lib.RequestResponse) {
+	if req, ok := record.IsRequest(); ok {
+		if err := l.logRequest(record.ID, req, req.Body); err != nil {
+			l.logError(record.ID, "request", err)
+		}
+	}
+	if resp, ok := record.IsResponse(); ok {
+		if err := l.logResponse(record.ID, resp, resp.Body); err != nil {
+			l.logError(record.ID, "response", err)
+		}
+	}
+}
+
+// Log takes request and response data and tries to write them into files created by this logger.
+// Note: this method is used by "old" clients that are not based on the rest client of the core lib.
```

Comment:
nit: technically used by just the old rest client, not several. You could even link to the method that calls it